_includes/v22.2/start-in-docker/mac-linux-steps-single-node.md

```diff
@@ -0,0 +1,128 @@
+
+{{site.data.alerts.callout_danger}}
+Single-node clusters are not highly-available or fault-tolerant. They are not appropriate for production use.
+{{site.data.alerts.end}}
+
+- You can optionally set the following [Docker environment variables](https://docs.docker.com/engine/reference/commandline/run/#set-environment-variables--e---env---env-file) to create a database and user automatically and to set a password for the user.
+  - `COCKROACH_DATABASE`
+  - `COCKROACH_USER`
+  - `COCKROACH_PASSWORD`
+
+  To prevent data loss, the environment variables are ignored if the `/cockroach/cockroach-data` directory within the container is not empty. To re-initialize the default database, user, and password from the environment variable values, delete the contents of `/cockroach/cockroach-data` within the running container.
+
+- You can optionally mount a directory of initialization scripts into the `docker-entrypoint-initdb.d` directory within the container. These scripts are run after CockroachDB starts and after the database and user (if specified) have been created. The scripts run in the alphanumeric sort order imposed by your locale.
+
+This section shows how to start a single-node cluster that uses these features.
+
+### Step 1. Create Docker volumes for the node
+
+Cockroach Labs recommends that you store cluster data in a Docker volume rather than in the storage layer of the running container. Otherwise, if a Docker container is inadvertently deleted, its data is inaccessible.
+
+To create the [Docker volume](https://docs.docker.com/storage/volumes/) where the cluster will store its data:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+docker volume create roach-single
+~~~
+
+
+
+### Step 2. Start the cluster
+
+1. Start the cluster node.
+
+    The following command starts a single-node cluster that:
+ 
+    - Stores its data in the `roach-single` volume on the Docker host, which is mounted on the `/cockroach/cockroach-data` directory within the container.
+    - If the `/cockroach/cockroach-data` within the container is empty, the specified database, user, and password are created automatically. Instead of specifying each value directly by using the `-e` or `--env` flag, you can store them in a file on the Docker host. Use one key-value pair per line and set the `--env-file` flag to the file's path.
+    - Bind-mounts the `~/init-scripts` directory on the Docker host onto the `/docker-entrypoint-initdb.d` directory within the container. Initialization scripts stored in this directory are run after CockroachDB starts and the default database, user, and password are initialized.
+    - Accepts database client connections on IP address 172.18.0.3 (which resolves to the hostname `roach-single` within the Docker container) on port 26257.
+    - Accepts connections to the DB Console on IP address 172.18.0.3 (which resolves to the hostname `roach-single` within the Docker container) on port 8080.
+
```

Comment:
As illustrated in [the existing docker page](https://www.cockroachlabs.com/docs/stable/start-a-local-cluster-in-docker-mac.html#step-1-create-a-bridge-network), we need to create a docker network first.
In the following example, we have the IP and advertise address `172.18.0.3`, so we need to do
```
docker network create -d bridge --subnet=172.18.10.3/16 roachnet
```
Note that the subnet address can be customized.

Reply:
Since the existing doc says "Since you'll be running multiple Docker containers on a single host, with one CockroachDB node per container, you need to create what Docker refers to as a [bridge network](https://docs.docker.com/engine/userguide/networking/#/a-bridge-network)", I made the assumption that you don't need a Docker network, since you aren't running multiple Docker containers on a single host. Do I need to update the non-single-user doc to reflect a different reason that the Docker network is needed?

Reply:
You're right, we don't need to create a network in this case 👍